tests/disaster-recovery/regional-dr/test_managed_cluster_node_failure.py

```diff
@@ -0,0 +1,204 @@
+    ],
+)
+class TestManagedClusterNodeFailure:
+    """
+    Test to verify failure of node hosting important pods of different
+    components in Regional-DR solution do not impact the mirroring
+
+    """
+
+    label_ns_map = {
+        "ramen-dr-cluster-operator": {
+            "label": constants.RAMEN_DR_CLUSTER_OPERATOR_APP_LABEL,
+            "namespace": constants.OPENSHIFT_DR_SYSTEM_NAMESPACE,
+        },
+        "rbd-mirror": {
+            "label": constants.RBD_MIRROR_APP_LABEL,
+            "namespace": constants.OPENSHIFT_STORAGE_NAMESPACE,
+        },
+        "submariner-gateway": {
+            "label": constants.SUBMARINER_GATEWAY_ACTIVE_LABEL,
+            "namespace": constants.SUBMARINER_OPERATOR_NAMESPACE,
+        },
+    }
+
+    @pytest.fixture(autouse=True)
+    def teardown(self, request):
+        def finalizer():
+            for cluster in get_non_acm_cluster_config():
+                config.switch_ctx(cluster.MULTICLUSTER["multicluster_index"])
+                logger.info("Checking for Ceph Health OK")
+                ceph_health_check(tries=40, delay=60)
+
+        request.addfinalizer(finalizer)
+
+    def test_single_managed_cluster_node_failure(
+        self, resource_on_node, dr_workload, nodes_multicluster, node_restart_teardown
+    ):
+        """
+        Tests to fail the node where the given pod is hosted in each managed cluster separately
+        verify the pod is rescheduled on a healthy node and mirroring is working as expected.
+
+        """
+        dr_workload(num_of_subscription=1)
+        managed_clusters = get_non_acm_cluster_config()
+
+        for cluster in managed_clusters:
+            index = cluster.MULTICLUSTER["multicluster_index"]
+            config.switch_ctx(index)
+            logger.info(
+                f"Inducing node failure in cluster: {cluster.ENV_DATA['cluster_name']}"
+            )
+            resource = Pod(
+                **get_pods_having_label(
+                    label=self.label_ns_map[resource_on_node]["label"],
+                    namespace=self.label_ns_map[resource_on_node]["namespace"],
+                )[0]
+            )
+            resource_node = get_pod_node(resource)
+            logger.info(
+                f"Stopping {resource_node.name} where {resource.name} is hosted"
+            )
+            nodes_multicluster[index].stop_nodes([resource_node])
+            wait_for_nodes_status(
+                node_names=[resource_node.name], status=constants.NODE_NOT_READY
+            )
+
+            # Wait for pod to be rescheduled. For submariner-gateway, use sleep as it doesn't get rescheduled
+            if resource_on_node == "submariner-gateway":
+                logger.info("Waiting for 300 seconds before starting the node")
+                time.sleep(300)
+            else:
+                # Wait for pod to reach terminating state or to be deleted
+                assert wait_for_pods_to_be_in_statuses(
```

Comment:
The comment says we are checking multiple statuses but I could see only one condition for Terminating.